solutions/cf-702C.mdx

```diff
@@ -188,6 +188,73 @@ int main() {
 
 </CPPSection>
 	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class CellularNetwork2 {
+	static final int MOD = 1000000007;
+	static final int MAXIMUM_SIZE = 100000;
+
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+
+		int n = io.nextInt();
+		int m = io.nextInt();
+		int[] dist = new int[MAXIMUM_SIZE];
+
+		Arrays.fill(dist, 2 * MOD);
+
+		int[] a = new int[n];
+		int[] b = new int[m];
+
+		for (int x = 0; x < n; x++) {
+			a[x] = io.nextInt();
+		}
+
+		for (int x = 0; x < m; x++) {
+			b[x] = io.nextInt();
+		}
+
+		int pointer1 = 0;
+
+		// pointer going left to right
+		for (int i = 0; i < n; i++) {
+			while (pointer1 < (m - 1) && b[pointer1 + 1] <= a[i]) {
+				// keep on incrementing
+				pointer1++;
+			}
+			dist[i] = Math.min(dist[i], Math.abs(a[i] - b[pointer1]));
+		}
+
+		pointer1 = m - 1;
+
+		// pointer going right to left
```

Comment:
these comments don't really explain anything tbh
i mean what does the pointer going to the left *do*

Reply:
i don't know, if i did every problem in a way where I tried to understand everything, adding support to java to every problem would take forever

Reply:
Well you probably want to try to make sure you understand the problem so you can help people who are trying to read through your code understand what everything is doing. Imo quality > quantity